app/models/incomeSourceDetails/incomeSourceIds/IncomeSourceIdHash.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.incomeSourceDetails.incomeSourceIds
+
+import models.incomeSourceDetails.incomeSourceIds.IncomeSourceIdHash.mkFromIncomeSourceId
+
+import scala.util.Try
+
+class IncomeSourceIdHash private(val hash: String) extends AnyVal {
+  override def toString: String = s"IncomeSourceIdHash: $hash"
+
+  def oneOf(ids: List[IncomeSourceId]): Option[IncomeSourceId] = {
+    ids.find(id => {
+      val hashId = mkFromIncomeSourceId(id)
+      hashId.hash == this.hash
+    })
+  }
+}
+
+object IncomeSourceIdHash {
+  def mkFromIncomeSourceId(id: IncomeSourceId): IncomeSourceIdHash = {
```

Comment:
if you can keep original naming, here type not relevant in the naming conversion

Reply:
I can change it back. I thought this was better as we have 2 different methods to make the IncomeSourceHash with different inputs 🤷

Reply:
you can look at this as a suggestion, I don't mind new naming convention as long as this would be consistent in other part of code base